pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+)
+
+type tlsValidationResult struct {
+	Valid   bool
+	Message string
+}
+
+// validateTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
+// If TLS is correctly configured, the function will return a result where result.Valid = true.
+// If it's incorrectly configured the result will instead have result.Valid = false and a message for the user in result.Message.
+func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDB) (tlsValidationResult, error) {
+	if !mdb.Spec.Security.TLS.Enabled {
+		return tlsValidationResult{Valid: true}, nil
+	}
+
+	r.log.Info("Ensuring TLS is correctly configured")
+
+	// Ensure CA ConfigMap exists
+	var caConfigMap corev1.ConfigMap
+	if err := r.client.Get(context.TODO(), mdb.TLSConfigMapNamespacedName(), &caConfigMap); err != nil {
```

Comment:
maybe here we can use the `configmap` package

Reply:
Good points. Switched to configmap and secret now!